src/Npgsql/NpgsqlConnectionStringBuilder.cs

```diff
@@ -1599,6 +1599,30 @@ internal static bool TrySplitHostPort(ReadOnlySpan<char> originalHost, [NotNullW
             return false;
         }
 
+        internal static bool IsUnixSocket(string host, int port, [NotNullWhen(true)] out string? socketPath, bool replaceForAbstract = true)
+        {
+            socketPath = null;
+            if (string.IsNullOrEmpty(host))
+                return false;
+
+            var isPathRooted = Path.IsPathRooted(host);
+
+            if (host[0] == '@')
+            {
+                if (replaceForAbstract)
+                    host = $"\0{host.Substring(1)}";
```

Comment:
What's with the `\0`, is this somehow required when opening an abstract unix socket? Any docs?

Reply:
https://man7.org/linux/man-pages/man7/unix.7.html

> abstract: an abstract socket address is distinguished (from a pathname socket) by the fact that sun_path[0] is a null byte ('\0').

Reply:
:open_mouth: 

Thanks...